dashboard/lib/policies/gender.rb

```diff
@@ -1,19 +1,122 @@
 class Policies::Gender
-  MALE_REGEX = /(^m$|\bmale|boy|guy|\bm(a|e)n|masculin|him|\bhe\b|hombre|dude|\bmail)/
-  FEMALE_REGEX = /(^f$|\bfemal|girl|gal|\bwom(a|e)n|fem(e|i)nin|she|\bher|mujer|\bfemail|\bfem)/
-  NON_BINARY_REGEX = /(^n$|they|them|non((\ |-)?)binary|inter(\ )?sex|gender(\ )?fluid|inter(\ )?gender|agender|boyflux|\btrans|\bit)/
+  module MatchingPatterns
```

Comment:
I like the refactor, but I'm unsure if it makes sense to keep the matching pattern in the Polices since it only contains constants.
I don't think we have a `lib/constants` directory.
I'm not sure what would be a better place to locate these variables.

Reply:
[`Policy Object`](https://dev.to/renatamarques97/design-patterns-with-ruby-on-rails-part-1-introduction-and-policy-object-1c37) is a design pattern used to manage permissions, roles, and policies. It determines whether an entity is allowed to perform a specific action but does not execute any actions itself. For tasks like normalization, we should use Service Objects.

Reply:
You can also read more about seven fundamental Rails patterns in this article: [7 Ways to Decompose Fat ActiveRecord Models](https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models)

Reply:
Alternatively, we could use a [Rails Concern](https://www.writesoftwarewell.com/how-rails-concerns-work-and-how-to-use-them/#what-is-a-concern) to encapsulate gender-specific business logic for the User model:
### `/dashboard/app/models/concerns/user/genderable.rb`
```ruby
# frozen_string_literal: true

module User::Genderable
  extend ActiveSupport::Concern

  MALE_GENDER_REGEXES = [...].freeze
  FEMALE_GENDER_REGEXES = [...].freeze
  NON_BINARY_GENDER_REGEXES = [...].freeze

  GENDERS = [
    NON_BINARY_GENDER = 'n',
    MALE_GENDER       = 'm',
    FEMALE_GENDER     = 'f',
    OTHER_GENDER      = 'o',
  ].freeze

  class_methods do
    def normalize_gender(raw_gender)
      return if raw_gender.blank?

      case raw_gender.strip.downcase
      when *NON_BINARY_GENDER_REGEXES then NON_BINARY_GENDER
      when *MALE_GENDER_REGEXES then MALE_GENDER
      when *FEMALE_GENDER_REGEXES then FEMALE_GENDER
      else OTHER_GENDER
      end
    end
  end

  included do
    # This callback can be removed
    # before_validation on: :create, if: -> {gender.present?} do
    #   self.gender = gender
    # end

    before_validation on: %i[create update], if: -> {gender_teacher_input.present? && will_save_change_to_attribute?('properties')} do
      self.gender = gender_teacher_input
    end

    before_validation on: %i[create update], if: -> {gender_student_input.present? && will_save_change_to_attribute?('properties')} do
      self.gender = gender_student_input.strip!
    end

    validates :gender_student_input, length: {maximum: 50}, no_utf8mb4: true
    validates :gender_teacher_input, no_utf8mb4: true
    validates :gender, inclusion: {in: GENDERS}, allow_blank: true
  end

  def gender=(value)
    super self.class.normalize_gender(value)
  end
end
```